test/Npgsql.Tests/CommandTests.cs

```diff
@@ -301,7 +301,9 @@ public async Task CancelAsyncImmediately()
             using var cmd = conn.CreateCommand();
             cmd.CommandText = "SELECT 1";
 
-            Assert.ThrowsAsync<OperationCanceledException>(async () => await cmd.ExecuteScalarAsync(cts.Token));
+            var t = cmd.ExecuteScalarAsync(cts.Token);
+            Assert.That(t.IsCompleted, Is.EqualTo(true)); // checks, if a query has completed synchronously
```

Comment:
:+1:

Nit: 

```suggestion
            Assert.That(t.IsCompleted, Is.True); // checks, if a query has completed synchronously
```

Reply:
You're just in time - I've almost merged 😝